GageStatsAgent/GageStatsAgent.cs

```diff
@@ -208,13 +208,26 @@ public IQueryable<String> GetRoles()
 
         #endregion 
         #region Station
-        public IQueryable<Station> GetStations()
+        public IQueryable<Station> GetStations(List<string> stationTypeList = null, List<string> agencyList = null, int page = 1)
         {
-            return Select<Station>().Include(s=>s.StationType).Include(s=>s.Agency);
+            // set number of items to skip/return for pagination
+            var numItemsReturned = 50;
+            var skip = (page - 1) * numItemsReturned;
+
+            var query = this.Select<Station>();
+            // if filters, apply them before returning query
+            if (stationTypeList.Any() == true)
+                query = query.Where(st => stationTypeList.Contains(st.StationTypeID.ToString()) || stationTypeList.Contains(st.StationType.Code.ToLower()));
+            if (agencyList.Any() == true)
+                query = query.Where(st => agencyList.Contains(st.AgencyID.ToString()) || agencyList.Contains(st.Agency.Code.ToLower()));
+
+            if (query.Count() > numItemsReturned)
+                sm("Returning page " + page + " of " + (query.Count() / numItemsReturned + 1) + ".");
+            return query.OrderBy(s => s.ID).Skip(skip).Take(numItemsReturned);
```

Comment:
just one more comment regarding removing pagination from agent. Moving the method to the service controller provides the agent to be more versatile for other methods, such as GetStation(ID). you will still be able to use the same method above, this allows the developer to follow the DRY rule. I know there isn't much in there now, but if the developer wants to enforce some criteria  on stations (such as ordering by station name) or something like that you would only have to add the code in the main getStation method, instead of now in both the GetStation and getStation(ID) methods